src/extension.ts

```diff
@@ -122,15 +122,15 @@ async function serverOpenConfig(tomcat: TomcatController, tomcatItem ?: TomcatSe
 
 async function createServer(tomcat: TomcatController): Promise<string> {
     const ui: VSCodeUI = new VSCodeUI();
-    const tomcatPath: string = await selectFolder(ui, Utility.localize('tomcatExt.selectinstllpath', 'Select Tomcat Installation Path'));
+    const tomcatPath: string = await selectFolder(ui, Utility.localize('tomcatExt.selectinstallpath', 'Select Tomcat Installation Path'));
     const serverName: string = path.basename(tomcatPath);
 
     if (tomcat.getTomcatServer(serverName)) {
-        return Promise.reject(new Error(
-            Utility.localize('tomcatExt.alreadyexist', 'This tomcat server exists in the workspace, abort creation')));
+        vscode.window.showInformationMessage(
```

Comment:
vscode.window.showInformationMessage is an async function, you will lose that promise, is it by design? and the return `${serverName};${tomcatPath}`; clause is executed before the confirm/ok button is clicked.

Reply:
Yes, there is no need to wait for the ok/confirm button clicked to proceed the operation. Just go ahead to run/debug on the same server.

Reply:
The current implementation will reject the operations and let the caller to handle the error. If there is different UI requirement. I would like to suggestion reject with different types of error

Reply:
When user create the same server to run the war package, or debug war package on it, we could think that it is the server that he/she wants to operate on. And the server already exists, so from user's perspective, it is more acceptable that the operation continues instead of being interrupted by an error.

Reply:
Took @jdneo 's advice, if user are creating the same server to deploy a war package to, pop up a warning message and offer a option to proceed or cancel.